code/modules/reagents/chemistry/reagents/water.dm

```diff
@@ -20,10 +20,65 @@
 	drink_desc = "The father of all refreshments."
 
 /datum/reagent/water/reaction_mob(mob/living/M, method=TOUCH, volume)
-// Put out fire
 	if(method == TOUCH)
+		// Put out fire
 		M.adjust_fire_stacks(-(volume / 10))
 		M.ExtinguishMob()
+		if(ishuman(M))
```

Comment:
Any suggestion to fix it? You commented on it earlier, but I mentioned there wasn't much way to do this otherwise.